test/packetimpact/tests/fin_wait2_reset_test.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2020 The gVisor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fin_wait2_reset_test
+
+import (
+	"golang.org/x/sys/unix"
+	"gvisor.dev/gvisor/pkg/tcpip/header"
+	tb "gvisor.dev/gvisor/test/packetimpact/testbench"
+	"testing"
+	"time"
+)
+
+func TestFinWait2_RST(t *testing.T) {
+	dut := tb.NewDUT(t)
+	listenFd, remotePort := dut.CreateListener(unix.SOCK_STREAM, unix.IPPROTO_TCP, 1)
+	defer dut.Close(listenFd)
+	conn := tb.NewTCPIPv4(t, dut, tb.TCP{DstPort: &remotePort}, tb.TCP{SrcPort: &remotePort})
+	defer conn.Close()
+	conn.Handshake()
+
+	acceptFd, _ := dut.Accept(listenFd)
+	tv := unix.Timeval{Sec: 1, Usec: 0}
+	dut.SetSockOptTimeval(acceptFd, unix.SOL_TCP, unix.TCP_LINGER2, &tv)
+	// Initiate Termination of Connection from DUT - Triggers FIN-ACK towards Test Bench.
+	dut.Close(acceptFd)
+
+	// Expecting FIN-ACK from DUT.
+	if gotOne := conn.Expect(tb.TCP{Flags: tb.Uint8(header.TCPFlagFin | header.TCPFlagAck)}, time.Second); gotOne == nil {
+		t.Fatal("expected a FIN-ACK within 1 second but got none")
+	}
+
+	// Send ACK to DUT.
+	conn.Send(tb.TCP{Flags: tb.Uint8(header.TCPFlagAck)})
+
```

Comment:
At this point, the DUT is supposed to try to receive data.  The DUT should be calling receive on the socket.  A new protobuf message will be needed to do this.  There are already many of them in posix_server.cc, in dut.go.  We just need to add a new one for making a call to receive.

Reply:
[GES] We confirmed, Ack sent from testbench reached DUT.
Is this to confirm Ack message indeed received by DUT. Also In dut.rcv(), if it receives Ack need to send a udp message as ‘ack received’ and if not send udp message ‘ack NOT received’ over control path. Please confirm. Any example code of this kind will be helpful

Reply:
It should be dut.recv(), not dut.rcv, so that it will match the linux name.  I think that is easiest.

Linux `recv` arguments are sockfd, buf, len, and flags.  So the protobuf version should use the same, except that when sending the command to the DUT, it should be just len.  And when receiving the response, it should be just the return value, errno, and buf.

Can you explain the part about the ACK.  Why do we need to send the "ack received" and the "ack no received" over the control path?  Is that in the original test somewhere?  I don't understand.

Reply:
[GES] Ok. Got it.
Please ignore my query on sending 'ack received' ...